### PR TITLE
Doing some updates to the magpie

### LINKED ISSUE #6932: Upgrade modules in magpie and do a cleanup
### Area for Improvement (required)
Upgrade modules in magpie and do a cleanup like removing global variables especially context objects.

### Observed behavior (required)
Magpie Pod can sometimes be Terminated. By doing some cleanup and upgrading the Go modules, we can see if that happens again.

### Desired behavior (required)
* Magpie being up-to-date
* Cleanup

### Proposed Fix (optional)
* Upgrade the Go modules to the latest versions
* Cleanup

## Additional context

[AB#10706](https://dev.azure.com/azure-octo/e61041b4-555f-47ae-95b2-4f8ab480ea57/_workitems/edit/10706)

### PR DESCRIPTION
**Pull request by ytimocin**
# Description

**_Opened this PR not from the fork because there are some actions that I would like to test and can't be triggered from a fork._**

1. Updating some packages
2. Changing the way we append string to an array
3. Adding missing context

## Type of change
- This pull request is a minor refactor, code cleanup, test improvement, or other maintenance task and doesn't change the functionality of Radius (issue link optional).
Fixes: #6932 

### Summary
🐰📦🚀

Updated the magpiego bindings for RabbitMQ and servicebus to use context and improve performance and readability. Also upgraded some dependencies in `go.mod` to use the latest versions.

> _Oh we're the magpiego crew and we know what to do_
> _We import the context and we update the `go.mod`_
> _We heave away and haul away with RabbitMQ_
> _And we make the servicebus go faster and smoother too_

### Walkthrough
*  Import context package and use context.Background() to create a context for RabbitMQ operations ([link](https://github.com/radius-project/radius/pull/6925/files?diff=unified&w=0#diff-7842d685b892c11c5740e574ffc57e49b46f65fec0bfa5180c931822b19f26edR4), [link](https://github.com/radius-project/radius/pull/6925/files?diff=unified&w=0#diff-7842d685b892c11c5740e574ffc57e49b46f65fec0bfa5180c931822b19f26edR53))
* Simplify message body conversion to string in GetMessage function for RabbitMQ binding ([link](https://github.com/radius-project/radius/pull/6925/files?diff=unified&w=0#diff-186427a868cc7f255398e6a829f74396e4ed3f1c16dbe593cfa9e1c313fe3a45L116-R115))
* Remove unused fmt package from servicebus.go file ([link](https://github.com/radius-project/radius/pull/6925/files?diff=unified&w=0#diff-186427a868cc7f255398e6a829f74396e4ed3f1c16dbe593cfa9e1c313fe3a45L5))
* Update go.mod file to use latest versions of mongo-driver, compress, crypto, and text packages ([link](https://github.com/radius-project/radius/pull/6925/files?diff=unified&w=0#diff-069ce70246ee93b05a3149affbf0f507f1a4482e8086c6ebb7dbeafb7df3f77dL17-R17), [link](https://github.com/radius-project/radius/pull/6925/files?diff=unified&w=0#diff-069ce70246ee93b05a3149affbf0f507f1a4482e8086c6ebb7dbeafb7df3f77dL33-R35), [link](https://github.com/radius-project/radius/pull/6925/files?diff=unified&w=0#diff-069ce70246ee93b05a3149affbf0f507f1a4482e8086c6ebb7dbeafb7df3f77dL41-R45))

### CODE REVIEW

File: test/magpiego/bindings/rabbitmq.go [hunk 1]
```diff
@@ -49,6 +50,7 @@ func RabbitMQBinding(envParams map[string]string) BindingStatus {
 		return BindingStatus{false, "Failed to declare a queue"}
 	}
 	msg := "Hello World!"
+	ctx := context.Background()
 	err = ch.PublishWithContext(
 		ctx,
```
Comment (kachawla) [inline test/magpiego/bindings/rabbitmq.go:55]: This is weird that ctx wasn't defined before.. how was it compiling?

**Review by kachawla — DISMISSED** [review]

File: test/magpiego/bindings/rabbitmq.go (hunk 1 above)
Comment (rynowak) [reply test/magpiego/bindings/rabbitmq.go:55]: Super wierd! Is this a global variable defined somewhere? If that's the case we should clean it up.

Comment (ytimocin) [reply test/magpiego/bindings/rabbitmq.go:55]: https://github.com/radius-project/radius/blob/main/test/magpiego/bindings/mongo.go#L12

**Review by kachawla — DISMISSED** [review]

File: test/magpiego/bindings/rabbitmq.go (hunk 1 above)
Comment (rynowak) [reply test/magpiego/bindings/rabbitmq.go:55]: We should fix that. Doesn't have to be this PR.

Storing a context in a global variable like that is going to lead to a lot more confusion.

File: test/magpiego/go.mod
Comment (ytimocin) [inline test/magpiego/go.mod:1]: Upgrading all modules to keep things up-to-date

**Review by rynowak — APPROVED** [review]
Awesome, thanks for doing the cleanup.

**Review by kachawla — APPROVED** [review]

**Pull request merged**